### PR TITLE
Adding tests for users, courses and fileexchange

### LINKED ISSUE #47: Tests schreiben für das Backend

### PR DESCRIPTION
**Pull request by gitrekkkt**
Closes issue #47

### CODE REVIEW

File: eCourse_backend/Readme.md [hunk 1]
```diff
@@ -1,22 +1,38 @@
...
\ No newline at end of file
+# Running tests
+
+```
+./manage.py test
+```
+
+## Coverage
+
+Install with pip
```
Comment (phie73) [inline eCourse_backend/Readme.md]: can it be included in the requirements.txt? Or is there a need to run pip install coverage separately?

File: eCourse_backend/courses/tests.py [hunk 2]
```diff
@@ -1,4 +1,124 @@
...
+        self.assertTrue(Course.objects.filter(name='ChangedCourse').exists())
+
+        response = self.client.post(
+            reverse('detailed_course', args=(my_id,)))
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        """
+        response = self.client.post(
+            reverse('course_overview', kwargs={'page': 1}))
+        self.assertEqual(response.status_code, HTTPStatus.OK)
+        """
```
Comment (phie73) [inline eCourse_backend/courses/tests.py]: can this comment be removed?

File: eCourse_backend/fileexchange/tests.py [hunk 3]
```diff
@@ -1,19 +1,285 @@
...
+        # create an exercise to do stuff with in the tests
+        """
+        exercise_form = {
+            'course': Course.objects.get(name='test_course_fe'),
+            'description': '420 lodern Sie es!',
+            'start_time': '2020-08-08 14:00:00',
+            'work_time_duration': '01:00:00',
+            'submission_deadline': '2020-08-08 15:15:00',
+        }
+        """
```
Comment (phie73) [inline eCourse_backend/fileexchange/tests.py]: think this can be removed

File: eCourse_backend/fileexchange/tests.py [hunk 4]
```diff
@@ -1,19 +1,285 @@
...
+
+        my_id = Exercise.objects.get(description='420 lodern Sie es!').id
+        response = self.client.get(reverse('alter_exersice', kwargs={
+                                   'id': my_id}))
+        self.assertEqual(response.status_code, HTTPStatus.FOUND)
+        self.assertRedirects(self.client.get(reverse('alter_exersice', kwargs={
+            'id': my_id})),
+            '/accounts/login/?next=/file_exchange/exercises/alter/1')
+
+        # TODO upload_file, download_file
```
Comment (phie73) [inline eCourse_backend/fileexchange/tests.py]: please remove todo

**Review by phie73 — CHANGES_REQUESTED** [review]

File: eCourse_backend/Readme.md (hunk 1 above)
Comment (gitrekkkt) [reply eCourse_backend/Readme.md]: Coverage is not needed to run the app and therefore not a real _requirement_. But I can include it

File: eCourse_backend/fileexchange/tests.py (hunk 3 above)
Comment (gitrekkkt) [reply eCourse_backend/fileexchange/tests.py]: done

**Review by phie73 — APPROVED** [review]

**Review by codingHahn — APPROVED** [review]
LGTM

**Pull request merged**